src/tr2/game/objects/creatures/trex.c

```diff
@@ -32,3 +65,104 @@ void TRex_Setup(void)
 
     g_AnimBones[obj->bone_idx + 40] |= BF_ROT_Y;
 }
+
+void __cdecl TRex_Control(const int16_t item_num)
+{
+    if (!Creature_Activate(item_num)) {
+        return;
+    }
+
+    ITEM *const item = Item_Get(item_num);
+    CREATURE *const creature = item->data;
+
+    int16_t tilt = 0;
+    int16_t angle = 0;
+    int16_t head = 0;
+
+    if (item->hit_points > 0) {
+        AI_INFO info;
+        Creature_AIInfo(item, &info);
+        if (info.ahead) {
+            head = info.angle;
+        }
+
+        Creature_Mood(item, &info, MOOD_ATTACK);
+        angle = Creature_Turn(item, creature->maximum_turn);
+        if (item->touch_bits != 0) {
+            g_LaraItem->hit_points -= item->current_anim_state == TREX_STATE_RUN
+                ? TREX_TRAMPLE_DAMAGE
+                : TREX_TOUCH_DAMAGE;
+        }
+
+        creature->flags = creature->mood != MOOD_ESCAPE && !info.ahead
+            && info.enemy_facing > TREX_FRONT_ARC
```

Comment:
I'm wondering if this should be `-TREX_FRONT_ARC` - currently the t-rex tends to just circle Lara.

Reply:
Great catch! That was a regression I introduced. Thank you.